### PR TITLE
MNT Use full flake8 rather than flake8_diff.sh

### PR DESCRIPTION
**Pull request by lesteve**
Easier to understand, easier to run locally, `flake8 --diff` has been removed in `flake8>=6` see #1383. There are not that many flake8 violations to fix.

### CODE REVIEW

File: doc/conf.py [hunk 1]
```diff
@@ -21,7 +21,7 @@
 # If your extensions are in another directory, add it here. If the directory
 # is relative to the documentation root, use os.path.abspath to make it
 # absolute, like shown here.
-#sys.path.append(os.path.abspath('.'))
+# ys.path.append(os.path.abspath('.'))
```
Comment (tomMoral) [inline doc/conf.py:24]: typo

File: joblib/test/test_func_inspect_special_encoding.py [hunk 2]
```diff
@@ -7,4 +7,3 @@ def big5_f():
     """
     # ����
```
Comment (tomMoral) [inline joblib/test/test_func_inspect_special_encoding.py:8]: Did not knew we had such comments 😅

File: joblib/test/test_memory.py [hunk 3]
```diff
@@ -303,9 +300,9 @@ def name_collision(x):
 def test_memory_warning_lambda_collisions(tmpdir):
     # Check that multiple use of lambda will raise collisions
     memory = Memory(location=tmpdir.strpath, verbose=0)
-    a = lambda x: x
+    a = lambda x: x  # noqa: 731
     a = memory.cache(a)
```
Comment (tomMoral) [inline joblib/test/test_memory.py:304]: maybe just use :

```suggestion
     a = memory.cache(lambda x: x)
```

**Review by tomMoral — APPROVED** [review]
I think this is a nice idea so it will be easier to maintain in the long run.
A couple of comments but otherwise LGTM

File: joblib/test/test_func_inspect_special_encoding.py (hunk 2 above)
Comment (lesteve) [reply joblib/test/test_func_inspect_special_encoding.py:8]: Me neither, live and learn every day :wink:

File: doc/conf.py [hunk 4]
```diff
@@ -62,7 +62,7 @@
 source_suffix = '.rst'
 
 # The encoding of source files.
-#source_encoding = 'utf-8'
+# ource_encoding = 'utf-8'
```
Comment (tomMoral) [inline doc/conf.py:65]: same typo

File: doc/conf.py [hunk 5]
```diff
@@ -82,35 +82,35 @@
 
 # The language for content autogenerated by Sphinx. Refer to documentation
 # for a list of supported languages.
-#language = None
+# anguage = None
```
Comment (tomMoral) [inline doc/conf.py:85]: for all the comments

File: joblib/test/test_memory.py [hunk 6]
```diff
@@ -303,9 +300,8 @@ def name_collision(x):
 def test_memory_warning_lambda_collisions(tmpdir):
     # Check that multiple use of lambda will raise collisions
     memory = Memory(location=tmpdir.strpath, verbose=0)
-    a = lambda x: x
-    a = memory.cache(a)
-    b = lambda x: x + 1
+     a = memory.cache(lambda x: x)
+    b = lambda x: x + 1  # noqa: 731
     b = memory.cache(b)
```
Comment (tomMoral) [inline joblib/test/test_memory.py:305]: I would use the same

```suggestion
    b = memory.cache(lambda x: x + 1)
```

**Review by tomMoral — COMMENTED** [review]
a few more typo

File: joblib/test/test_memory.py [hunk 7]
```diff
@@ -303,10 +300,8 @@ def name_collision(x):
 def test_memory_warning_lambda_collisions(tmpdir):
     # Check that multiple use of lambda will raise collisions
     memory = Memory(location=tmpdir.strpath, verbose=0)
-    a = lambda x: x
-    a = memory.cache(a)
-    b = lambda x: x + 1
-    b = memory.cache(b)
+     a = memory.cache(lambda x: x)
```
Comment (tomMoral) [inline joblib/test/test_memory.py:303]:
```suggestion
    a = memory.cache(lambda x: x)
```

**Review by tomMoral — APPROVED** [review]
LGTM, the failure does not seem related.
Merging and hoping this won't cause too much troubles with other PRs.

**Pull request merged**